File: Data Structures & Algorithms/two-integer-sum/submission-6.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for idx, num in enumerate(nums):
            matching = target - num
            if matching in seen:
                return [seen[matching], idx]
            seen[num] = idx
        return []

        

File: Data Structures & Algorithms/sort-colors/submission-6.java
class Solution {
    public void sortColors(int[] nums) {
        int[] colors = new int[3];
        for(int num: nums){
            colors[num] += 1;
        }
        int idx = 0;
        for(int colorIdx = 0; colorIdx < colors.length; colorIdx++){
            while(colors[colorIdx]-- > 0){
                nums[idx++] = colorIdx;
            }
        }
    }
}